kbchat/kbchat_test.go

```diff
@@ -0,0 +1,119 @@
+package kbchat
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+)
+
+type Bots struct {
+	Alice1   *OneshotOptions
+	Alice2   *OneshotOptions
+	Bob1     *OneshotOptions
+	Charlie1 *OneshotOptions
+}
+
+type Config struct {
+	Bots
+}
+
+func readAndParseConfig() (Config, error) {
+	var config Config
+	data, err := ioutil.ReadFile("test_config.yaml")
+	if err != nil {
+		return Config{}, err
+	}
+
+	err = yaml.Unmarshal(data, &config)
+	if err != nil {
+		return Config{}, err
+	}
+
+	return config, nil
+}
+
+func randomTempDir() (string, error) {
+	bytes := make([]byte, 16)
+	_, err := rand.Read(bytes)
+	if err != nil {
+		return "", err
+	}
+	dir := path.Join(os.TempDir(), "keybase_bot_"+hex.EncodeToString(bytes))
+	return dir, nil
+}
+
+func whichKeybase() (string, error) {
+	cmd := exec.Command("which", "keybase")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
```

Comment:
[`cmd.Output()`](https://golang.org/pkg/os/exec/#Cmd.Output) is probably the more idiomatic way of doing this

Reply:
oh yeah, I didn't know that existed!